infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -1041,6 +1052,33 @@ public LogHeader getLogHeader() {
     }
 
     /**
+     * Fetches the set of addresses known to the segment handler given a start and end address.
+     *
+     * @param startAddress start address specifying the range.
+     * @param endAddress end address specifying the range.
+     * @return Set of known addresses.
+     */
+    @Override
+    public Set<Long> getKnownAddressesInRange(long startAddress, long endAddress) {
+
+        long startSegment = startAddress / RECORDS_PER_LOG_FILE;
+        long endSegment = endAddress / RECORDS_PER_LOG_FILE;
+
+        Set<Long> knownAddressSet = new HashSet<>();
+        for (long segmentIndex = startSegment; segmentIndex <= endSegment; segmentIndex++) {
+            String filePath = logDir + File.separator + segmentIndex + ".log";
+
+            // Filters all addresses in range.
```

Comment:
writeChannels is an in-memory map, it can be empty when the logunit restarts, or can be trimmed when a prefix trim happens, which seems to violate the contract.